packages/edit-site/src/components/save-hub/index.js

```diff
@@ -32,29 +55,106 @@ export default function SaveHub() {
 		[]
 	);
 
+	const {
+		editEntityRecord,
+		saveEditedEntityRecord,
+		__experimentalSaveSpecifiedEntityEdits: saveSpecifiedEntityEdits,
+	} = useDispatch( coreStore );
+
 	const disabled = isSaving || ( ! isDirty && ! isPreviewingTheme() );
 
+	// if we have only one unsaved change and it matches current context, we can show a more specific label
+	let dirtyLocal = null;
+
+	if ( countUnsavedChanges === 1 ) {
+		// if we are on global styles
+		if ( params.path?.includes( 'wp_global_styles' ) ) {
+			dirtyLocal = dirtyRecords.find(
+				( record ) => record.name === 'globalStyles'
+			);
+		}
+		// if we are on pages
+		else if ( !! params.postId ) {
+			dirtyLocal = dirtyRecords.find(
+				( record ) =>
+					record.name === params.postType &&
+					String( record.key ) === params.postId
+			);
+		}
+	}
+
+	let label = dirtyLocal
+		? __( 'Save' )
+		: sprintf(
+				// translators: %d: number of unsaved changes (number).
+				_n(
+					'Review %d change…',
+					'Review %d changes…',
+					countUnsavedChanges
+				),
+				countUnsavedChanges
+		  );
+
+	if ( isSaving ) {
+		label = __( 'Saving' );
+	}
```

Comment:
Won't `SaveButton` do this as well?

https://github.com/WordPress/gutenberg/pull/50567/files#diff-4b7429763be74ad657ec888b6a33a0bbdbd8fcc38b9cae2a6d116eab19a4d0adR42-R44

Reply:
It does but we also need it for the direct saving button